### PR TITLE
cPanel patches

### PR DESCRIPTION
**Pull request by atoomic**
bunch of patches used by cPanel to tweak perl

### CODE REVIEW

**Comment by atoomic** [comment]
```
Test Summary Report
-------------------
op/closure.t                                                     (Wstat: 0 Tests: 282 Failed: 1)
  Failed test:  281
op/incfilter.t                                                   (Wstat: 65280 Tests: 0 Failed: 0)
  Non-zero exit status: 255
  Parse errors: No plan found in TAP output
porting/authors.t                                                (Wstat: 0 Tests: 14 Failed: 1)
  Failed test:  10
porting/customized.t                                             (Wstat: 0 Tests: 24 Failed: 1)
  Failed test:  17
porting/manifest.t                                               (Wstat: 0 Tests: 10439 Failed: 3)
  Failed tests:  10434, 10436-10437
../cpan/Filter-Util-Call/t/call.t                                (Wstat: 0 Tests: 32 Failed: 29)
  Failed tests:  2, 5-32
../cpan/Filter-Util-Call/t/rt_101033.t                           (Wstat: 65280 Tests: 0 Failed: 0)
  Non-zero exit status: 255
  Parse errors: No plan found in TAP output
../cpan/Filter-Util-Call/t/rt_54452-rebless.t                    (Wstat: 0 Tests: 2 Failed: 2)
  Failed tests:  1-2
Files=2618, Tests=1140116, 593 wallclock secs (150.66 usr 12.39 sys + 800.43 cusr 56.24 csys = 1019.72 CPU)
Result: FAIL
make: *** [test_harness] Error 37
```

File: ext/DynaLoader/DynaLoader_pm.PL [hunk 1]
```diff
@@ -468,8 +468,8 @@ sub dl_findfile {
 
         #  Only files should get this far...
         my(@names, $name);    # what filenames to look for
-        if (m:-l: ) {          # convert -lname to appropriate library name
```
Comment (xsawyerx) [inline ext/DynaLoader/DynaLoader_pm.PL]: Just do a s///

File: ext/DynaLoader/DynaLoader_pm.PL [hunk 2]
```diff
@@ -468,8 +468,8 @@ sub dl_findfile {
...
-            s/-l//;
+        if (index($_,'-l') == 0) {          # convert -lname to appropriate library name
+            substr($_,0,2,'');
             push(@names,"lib$_.$dl_so");
             push(@names,"lib$_.a");
```
Comment (xsawyerx) [inline ext/DynaLoader/DynaLoader_pm.PL]: Just do one push

File: cpan/Digest-SHA/lib/Digest/SHA.pm [hunk 3]
```diff
@@ -11,8 +11,10 @@ use integer;
 $VERSION = '5.98';
 
 require Exporter;
-require DynaLoader;
-@ISA = qw(Exporter DynaLoader);
+
+use XSLoader ();
+
+@ISA = qw(Exporter);
```
Comment (jkeenan) [inline cpan/Digest-SHA/lib/Digest/SHA.pm:17]: Shouldn't changes like this be submitted first on CPAN?

File: sv.c [hunk 4]
```diff
@@ -12377,7 +12377,7 @@ Perl_sv_vcatpvfn_flags(pTHX_ SV *const sv, const char *const pat, const STRLEN p
 		    elen = sizeof nullstr - 1;
 		}
 	    }
-	    else {
+	    else if (argsv) {
```
Comment (xsawyerx) [inline sv.c]: Yves says this shouldn't be a problem for pure perl

File: dist/Carp/lib/Carp.pm [hunk 5]
```diff
@@ -152,8 +152,7 @@ sub export_fail { shift; $Verbose = shift if $_[0] eq 'verbose'; @_ }
```
Comment (jkeenan) [inline dist/Carp/lib/Carp.pm:159]: Do you have any benchmarking data to support the claim of improved speed?

File: ext/B/B.pm [hunk 6]
```diff
@@ -6,11 +6,15 @@
 #      License or the Artistic License, as specified in the README file.
 #
 package B;
-use strict;
```
Comment (xsawyerx) [inline ext/B/B.pm]: Set $^H

File: dist/IO/lib/IO/Handle.pm [hunk 7]
```diff
@@ -368,7 +372,7 @@ sub fdopen {
 
     if (ref($fd) && "".$fd =~ /GLOB\(/o) {
```
Comment (xsawyerx) [inline dist/IO/lib/IO/Handle.pm]: "".$fd doesn't do what you think it does..?

=~has higher precedence; should have used parens

File: dist/IO/lib/IO/Handle.pm [hunk 8]
```diff
@@ -494,7 +498,7 @@ sub stat {
 ##
 
 sub autoflush {
-    my $old = new SelectSaver qualify($_[0], caller);
+    my $old = new SelectSaver Symbol::qualify($_[0], caller);
```
Comment (xsawyerx) [inline dist/IO/lib/IO/Handle.pm]: Remove indirect new

File: dist/Storable/Storable.pm [hunk 9]
```diff
@@ -97,14 +97,7 @@ XSLoader::load('Storable', $Storable::VERSION);
...
 
-sub CAN_FLOCK; my $CAN_FLOCK; sub CAN_FLOCK {
-	return $CAN_FLOCK if defined $CAN_FLOCK;
-	require Config; import Config;
-	return $CAN_FLOCK =
-		$Config{'d_flock'} ||
-		$Config{'d_fcntl_can_lock'} ||
-		$Config{'d_lockf'};
-}
+sub CAN_FLOCK { return 1 }
```
Comment (xsawyerx) [inline dist/Storable/Storable.pm]: Can we assume Flock based on $^O?

Comment (toddr) [reply dist/Storable/Storable.pm]: TL;DR; NO.

Comment (atoomic) [reply dist/Storable/Storable.pm]: note that this patch is obsoleted by ad2ec6b54c9968999b7c70984b5a203fa72fd540

**Comment by xsawyerx** [comment]
Can this PR be rebased based on the changes we've done or decisions we've made since? One comment from Todd shows there's at least one change that isn't necessary. Is that right?

**Comment by atoomic** [comment]
@xsawyerx, we are currently merging these patches as discussed and will rebase this branch in the next days once we are done with it. I need to process your remarks there and add them to our patches.

File: cpan/Digest-SHA/lib/Digest/SHA.pm (hunk 3 above)
Comment (toddr) [reply cpan/Digest-SHA/lib/Digest/SHA.pm:17]: Yes but the point of this PR was to show what we did to the tarball during build.

**Comment by toddr** [comment]
needed to re-base. Closing this case and opening as a new PR #34

**Pull request closed without merging**